src/web/CMakeLists.txt

```diff
@@ -1,41 +1,42 @@
-#
-# setup
-#
-set( WEB_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}  CACHE INTERNAL "" )
+add_executable(web-server)
 
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads REQUIRED)
 
-#
-# web server
-#
-if ( NOT ENABLE_WEB_SERVER )
-	return()
-endif ( NOT ENABLE_WEB_SERVER )
+target_sources(web-server PRIVATE
```

Comment:
As I responded in the above review, this isn't great.
> I specifically do not want to do this. When we add a new file, CMake won't know to regenerate the build system.
See the [official documentation](https://cmake.org/cmake/help/latest/command/file.html?highlight=CONFIGURE_DEPENDS#filesystem)

Reply:
I still agree with using `file(GLOB...) ` to get the list of files, although he does have the problem @vstumpf mentioned.

But I think if the problem did arise, I would personally use: `rm -rf * && cmake -G "Unix Makefiles" .. && make` in the `cbuild` directory, which is actually very low cost for me.

For a novice developer, `file(GLOB...)' is a lot less thought-provoking than having to find the corresponding CMakeLists.txt and add hpp and cpp respectively when adding files. Is much more thoughtless than having to find the corresponding CMakeLists.txt and find the corresponding location for hpp and cpp respectively.

Reply:
The chances of someone adding a file are pretty low, it's usually reserved for developers who (should) know what they're doing.
I'm imagining a user who updates rathena every couple months or so; if we change CMakeLists when a new file appears, IDEs (Visual Studio, vscode, etc) will automatically regenerate the build system. Otherwise, they will have to go in and clear the cache to get cmake to rebuild.
Ideally, finding the correct CMakeLists.txt won't be too hard, since it'll be in the same directory as the source file.